egui_extras/src/grid.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    layout::{CellSize, Layout, LineDirection},
+    sizing::Sizing,
+    Size,
+};
+use egui::{Response, Ui};
+
+/// The direction in which cells are positioned in the grid.
+///
+/// In a horizontal grid cells are positions from left to right.
+/// In a vertical grid cells are positions from top to bottom.
```

Comment:
This is still ambiguous. Consider this order:

```
0 1 2 3
4 5 6 7
```

This is positioned both from left to right _and_ from top to bottom. Is it "horizontal" or "vertical"?

In linear algebra the above would be called "row-major", while this would be "column-major":

```
0 2 4 6
1 3 5 7
```

Perhaps these are better terms? If this is indeed what `GridDirection` even means.

Reply:
A grid in my sense is only one-directional and can have sub-grids which have a different direction, so it is either
```
0 1 2 3
```

or

```
0
1
2
3
```

with subgrid it could look like this:

```
0 A 2 3
  B
  C
```

Reply:
Documentation is hard, I tried to write a better description and changed some struct and variant names to make it easier to understand.

Reply:
> A grid in my sense is only one-directional and can have sub-grids which have a different direction, so it is either
> 
> ```
> 0 1 2 3
> ```
> 
> or
> 
> ```
> 0
> 1
> 2
> 3
> ```

Oh, this was definitely surprising to me. I would not call a one-dimensional list a "grid".

Reply:
You could form a grid, but you are right, it was not a good name :roll_eyes: